generate_analytical_dataset.py

```diff
@@ -62,7 +52,40 @@ def main():
         spark.sql(src_hive_select_query).show()
 
 
-def decrypt(cek, kek, iv, ciphertext, keys_map):
+def get_dataframe_from_staging(adg_hive_select_query, spark):
+    df = spark.sql(adg_hive_select_query)
+    return df
+
+
+def get_spark_session():
+    spark = (
+        SparkSession.builder.master("yarn")
+            .config("spark.sql.parquet.binaryAsString", "true")
+            .appName("aws-analytical-dataset-generator")
+            .enableHiveSupport()
+            .getOrCreate()
+    )
+    return spark
+
+
+def retrieve_secrets():
+    secret_name = "${secret_name}"
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
+    client = session.client(service_name="secretsmanager")
+    response = client.get_secret_value(SecretId=secret_name)
+    response_dict = ast.literal_eval(response["SecretString"])
+    return response_dict
+
+
+def sanitize(decrypted, db_name, collection_name):
+    if ((db_name == "penalties-and-deductions" and collection_name == "sanction")
+            or (db_name == "core" and collection_name == "healthAndDisabilityDeclaration")
+            or (db_name == "accepted-data" and collection_name == "healthAndDisabilityCircumstances")):
+        decrypted = re.sub(r'(?<!\\)\\[r|n]', '', decrypted)
+    return decrypted.replace("$", "d_").replace("\u0000", "").replace("_archivedDateTime", "_removedDateTime").replace("_archived", "_removed")
```

Comment:
how come we are hard-coding collection names here?

Reply:
Only this collections needs those junk characters replacement. Similar implementation in the htme   as well https://github.com/dwp/hbase-to-mongo-export/blob/0ed6e6999cb9e1d098536516cce4720bab4d0dac/src/main/kotlin/app/batch/processor/SanitisationProcessor.kt#L35